src/Features/header/components/Cities/Cities.tsx

```diff
@@ -1,40 +1,66 @@
 import { FiNavigation } from 'react-icons/fi';
-import { ReactElement, useMemo, useCallback } from 'react';
+import { ReactElement, useMemo, useState, useEffect, useCallback } from 'react';
 
-import { useFetchCityListQuery } from '../Sidebar/SidebarSlice';
-import { LoadingSpinner } from '../../../../components/LoadingSpinner/LoadingSpinner';
+import { LoadingSpinner } from '../../../../components/LoadingSpinner';
+import { useLazyFetchCityListQuery } from '../Sidebar/SidebarSlice';
 import { CityList } from './Cities.types';
 
 export const Cities = (): ReactElement => {
-  const { data, isLoading } = useFetchCityListQuery('');
+  const [page, setPage] = useState(0);
+  const [offset] = useState(8);
+  const [trigger, result] = useLazyFetchCityListQuery();
+
+  const fetchMoreData = useCallback(() => {
+    trigger({ page, offset });
+    setPage((prev) => prev + 1);
+  }, [trigger, offset, page]);
 
   const handleCityClick = useCallback(() => {
-    return data;
-  }, [data]);
+    return result.data;
+  }, [result.data]);
+
+  useEffect(() => {
+    trigger({ page, offset });
+    setPage((prev) => prev + 1);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
trigger should be set to dependencies, I guess then you will be able to remove eslint ignore